src/nvim/getchar.c

```diff
@@ -2015,6 +2036,12 @@ static int vgetorpeek(int advance)
             else {
               int noremap;
 
+              // If this is a LANGMAP mapping, then we didn't record the keys
+              // at the start of the function and have to record them now.
+              if (keylen > typebuf.tb_maplen && (mp->m_mode & LANGMAP) != 0) {
+                gotchars(s, -1);
```

Comment:
I thought you might want to avoid the double iteration over characters in the expansion.

i.e. we already have a loop iterating over all characters in the buffer, so counting the length with `strlen()` is effectively iterating over the characters twice when it's not needed.

Honestly I wasn't sure which to use -- I guessed which would be preferred by others.